chaco/plot_containers.py

```diff
@@ -136,176 +124,7 @@ def get_preferred_size(self, components=None):
-            else:
-                each_size = int(avail_size / len(resizable_components))
-                for component in resizable_components:
-                    tmp = list(component.outer_bounds)
-                    tmp[ndx] = each_size
-                    new_bounds_dict[component] = tmp
-
-        # Loop over all the components, assigning position and computing the
-        # size in the other dimension and its position.
-        cur_pos = 0
-        for component in components:
-            if not self._should_layout(component):
-                continue
-
-            position = list(component.outer_position)
-            position[ndx] = cur_pos
-
-            bounds = new_bounds_dict.get(
-                component, list(component.outer_bounds)
-            )
-            cur_pos += bounds[ndx] + self.spacing
-
-            if (bounds[other_ndx] > size[other_ndx]) or (
-                other_dim in component.resizable
-            ):
-                # If the component is resizable in the other dimension or it exceeds the
-                # container bounds, set it to the maximum size of the container
-
-                position[other_ndx] = 0
-                bounds[other_ndx] = size[other_ndx]
-            else:
-                position[other_ndx] = 0
-                if align == "min":
-                    pass
-                elif align == "max":
-                    position[other_ndx] = size[other_ndx] - bounds[other_ndx]
-                elif align == "center":
-                    position[other_ndx] = (
-                        size[other_ndx] - bounds[other_ndx]
-                    ) / 2.0
-
-            component.outer_position = position
-            component.outer_bounds = bounds
-            component.do_layout()
-
-    ### Persistence ###########################################################
-
-    # PICKLE FIXME: blocked with _pickles, but not sure that was correct.
-    def __getstate__(self):
-        state = super(StackedPlotContainer, self).__getstate__()
-        if "stack_index" in state:
-            del state["stack_index"]
-        return state
-
-
-class HPlotContainer(StackedPlotContainer):
-    """
-    A plot container that stacks all of its components horizontally. Resizable
-    components share the free space evenly. All components are stacked from
-    according to **stack_order* in the same order that they appear in the
-    **components** list.
-    """
-
-    draw_order = Instance(list, args=(DEFAULT_DRAWING_ORDER,))
-
-    #: The order in which components in the plot container are laid out.
-    stack_order = Enum("left_to_right", "right_to_left")
-
-    #: The amount of space to put between components.
-    spacing = Float(0.0)
-
-    #: The vertical alignment of objects that don't span the full height.
-    valign = Enum("bottom", "top", "center")
-
-    _cached_preferred_size = Tuple(transient=True)
+        return stacked_preferred_size(self, components=components)
```

Comment:
```
        if self.resizable == "":
            self._cached_preferred_size = self.outer_bounds[:]
            return self.outer_bounds
```

isn't handled by enables `stacked_preferred_size`. If we can update the function in enable, after the next release of enable, we will be able to get rid of this method completely. both in this class and in `VPlotContainer`

Reply:
ref: https://github.com/enthought/enable/pull/778